mediar-ai/screenpipe
# feat(ui): add collapsible search history sidebar

# WIP
## description
Add collapsible search history sidebar to track and restore previous searches.

Fixes: #456 
/claim #456 

## type of change

## how to test

1. Make a few searches with different parameters
2. Check that searches are saved in the sidebar
3. Click on a previous search to restore its state
4. Verify that search parameters, results, and AI responses are restored
5. Test collapsing/expanding the sidebar
6. Test deleting search history items
7. Verify "new search" button clears the current state

Demo:

https://github.com/user-attachments/assets/6d7f4e8b-4d0c-4438-8b92-71cc2399159c



## checklist

## additional notes
- Uses localforage for persistent storage of search history
- Sidebar is collapsed by default
- Search history includes full search parameters, results and AI responses
- Added ability to restore complete search state including AI conversations
- Fixed all TypeScript errors and linting issues
